packages/app-form-builder/src/admin/components/FormEditor/Tabs/EditTab/EditFieldDialog/GeneralTab.tsx

```diff
@@ -32,6 +33,20 @@ const GeneralTab: React.FC<GeneralTabProps> = ({ field, form }) => {
         throw new Error("Please enter a unique Field ID");
     }, []);
 
+    const fieldIdValidator: Validator = (value: string | undefined) => {
```

Comment:
1. Let's move these two validators outside of the `GeneralTab` component. 
2. Let's assign `Validator` type to `uniqueFieldIdValidator`, the same way you did with `fieldIdValidator`. I know it's not part of your changes, still, let's sync it up.

Reply:
@adrians5j 
please add more details for a item 2) about `uniqueFieldIdValidator`.
It just that this validator is using not only a field value, but also uses `useFormEditor` hook to check for ID uniqueness.
So if it is outside a component with type `Validator` I have a difficulty to pass `useFormEditor` value in some way due to type restriction.